modules/nextflow/src/main/groovy/nextflow/extension/ChannelExtensionProvider.groovy

```diff
@@ -154,6 +163,14 @@ class ChannelExtensionProvider implements ExtensionProvider {
         def result = new HashSet<String>(30)
         def methods = clazz.getDeclaredMethods()
         for( def handle : methods ) {
+            // in a future only annotated methodS will be imported
+            if( handle.isAnnotationPresent(Operator)) {
+                def params=handle.getParameterTypes()
+                if( params.length>0 && isReadChannel(params[0]) ) {
```

Comment:
It would be better to return an error if `isReadChannel(params[0])` is false. In a second iteration, it would be cool to have an AST rule that check this during the compilation

Reply:
Let's add a warning instead if the operator is discovered via the deprecated snippet 

```
            if( params.length>0 && isReadChannel(params[0]) )
                result.add(handle.name)
```